packages/eslint-plugin/src/rules/no-invalid-void-type.ts

```diff
@@ -127,6 +127,35 @@ export default util.createRule<[Options], MessageIds>({
       }
     }
 
+    /**
+     * @brief checks if the generic type parameter extends void
+     */
+    function checkExtendedVoid(
+      node: TSESTree.TSVoidKeyword,
+      parentNode: TSESTree.TSTypeParameter,
+    ): void {
+      if (
+        parentNode.constraint?.type !== AST_NODE_TYPES.TSVoidKeyword ||
+        JSON.stringify(parentNode.default?.range) !== JSON.stringify(node.range)
```

Comment:
It seems like this is trying to check whether `parentNode.default` _is_ `node`. You can do that directly:

```suggestion
       parentNode.default?.range === node.range
```

Am I missing something?

If I'm understanding right, then the function body could look like:

```ts
if (parentNode.default !== node) {
  context.report({
    messageId: 'invalidVoidNotReturnOrGeneric',
    node,
  });
}
```

Reply:
Oops, just realized I left the `.range`s in there. Sorry! I'd meant to suggest:

```ts
parentNode.default === node
```

Reply:
Thank you for suggesting!
You're right, so I rewrite the function body as you suggested:
https://github.com/typescript-eslint/typescript-eslint/pull/5671/commits/be5c6f406011fc55a7d4793059099e358be4104c

https://github.com/typescript-eslint/typescript-eslint/pull/5671/commits/508ab8f52d939b10480b97a6cc1f47f85bd95768